src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -157,6 +158,93 @@ public boolean isPinned()
         return getTextChannel().clearReactionsById(getId());
     }
 
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(Emote emote)
+    {
+        return channel.removeReactionById(getId(), emote);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(Emote emote, User user)
+    {
+        Checks.notNull(emote, "Emote");
+        MessageReaction reaction = this.reactions.stream()
+            .filter(r -> r.getReactionEmote().getIdLong() == emote.getIdLong())
+            .findFirst().orElse(null);
+        Checks.check(reaction != null,
+            "Cannot remove reaction from a message that doesn't contain the reaction.");
+        return user != null ? reaction.removeReaction(user) : reaction.removeReaction();
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(String emote)
+    {
+        return channel.removeReactionById(getId(), emote);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(String emote, User user)
+    {
+        Emote e = this.reactions.stream()
+            .map(MessageReaction::getReactionEmote)
+            .filter(r -> r.isEmoji() && r.getEmoji().equals(emote))
+            .map(MessageReaction.ReactionEmote::getEmote)
+            .findFirst().orElse(null);
+        return removeReaction(e, user);
+    }
+
+    @Nonnull
+    @Override
+    public ReactionPaginationAction retrieveReactionUsers(Emote emote)
+    {
+        MessageReaction reaction = this.reactions.stream()
+            .filter(r -> r.getReactionEmote().getEmote().equals(emote))
+            .findFirst().orElse(null);
+        Checks.check(reaction != null,
+            "Cannot retrieve users for a reaction that doesn't exist.");
+        return reaction.retrieveUsers();
+    }
+
+    @Nonnull
+    @Override
+    public ReactionPaginationAction retrieveReactionUsers(String emote)
+    {
+        Emote e = this.reactions.stream()
+            .map(MessageReaction::getReactionEmote)
+            .filter(r -> r.isEmoji() && r.getEmoji().equals(emote))
+            .map(MessageReaction.ReactionEmote::getEmote)
+            .findFirst().orElse(null);
+        return retrieveReactionUsers(e);
+    }
+
+    @Override
+    public MessageReaction.ReactionEmote getReactionByName(String name)
+    {
+        return this.reactions.stream()
+            .map(MessageReaction::getReactionEmote)
+            .filter(r -> r.getName().equals(name))
+            .findFirst().orElse(null);
+    }
+
+    @Override
+    public MessageReaction.ReactionEmote getReactionById(String id)
+    {
+        return this.reactions.stream()
```

Comment:
Missing null checks, should be converted to long for better comparison

Reply:
Alright